src/traversing/findFilter.js

```diff
@@ -33,7 +33,7 @@ function winnow( elements, qualifier, not ) {
 	}
 
 	return jQuery.grep( elements, function( elem ) {
-		return ( indexOf.call( qualifier, elem ) > -1 ) !== not;
+		return ( indexOf.call( qualifier, elem ) > -1 ) !== not && elem.nodeType === 1;
```

Comment:
It may be a cheaper check on it's own, but it's a far less discriminating check in most use cases as it would only ever be `false` when using non-element nodes.

As far as I know, only the `.contents()` function can return non-element nodes, so not a very common situation.

Placing the `nodeType` check last would shortcut it in most cases.

Placing the `nodeType` check first would mean both checks would have to be evaluated in most cases.